### PR TITLE
add infra for widget creation

### PR DESCRIPTION
**Pull request by ronykoz**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
adding infra for widget creation to make the process easier.

This is still is not finalized, just want to see what you think

Example code for the line widget which is the most complex
```
tb_widget = LineWidget()
tb_widget.add_category(name='10-10-2020T', number=13, group='Rony')
tb_widget.add_category(name='10-11-2020T', number=13, group='Rony')
tb_widget.add_category(name='10-12-2020T', number=130, group='Rony')
tb_widget.add_category(name='10-13-2020T', number=1300, group='Rony')
tb_widget.add_category(name='10-10-2020T', number=100, group='Gal')
tb_widget.add_category(name='10-11-2020T', number=13, group='Gal')
tb_widget.add_category(name='10-12-2020T', number=344, group='Gal')
tb_widget.add_category(name='10-13-2020T', number=555, group='Gal')
return_results(tb_widget)
```

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -2666,6 +2666,10 @@ def return_results(results):
         demisto.results(results.to_context())
         return
 
+    if isinstance(results, BaseWidget):
+        demisto.results(results.publish())
```
Comment (Itay4) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2670]: why did we choose `publish` ?

Comment (ronykoz) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2670]: I'm up for suggestions

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -4035,3 +4039,115 @@ def update_integration_context(context, object_keys=None, sync=True):
...
+
+
+class BaseWidget:
+    @abstractmethod
+    def publish(self):
+        pass
+
+
+class TextWidget(BaseWidget):
+    def __init__(self, text):
```
Comment (Itay4) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:4051]: please add type hint here as well

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -4035,3 +4039,115 @@ def update_integration_context(context, object_keys=None, sync=True):
...
+        self.seconds = seconds
+
+    def publish(self):
+        return json.dumps([{
+            'name': self.name,
+            'data': [self.seconds]
+        }])
+
+
+class TableOrListWidget(BaseWidget):
```
Comment (Itay4) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:4140]: i'd split these to two

**Comment by ronykoz** [comment]
> what about `PieWidget`?

BarColumnPieWidget

**Comment by anara123** [comment]
@ronykoz looks good! great idea.

**Comment by glicht** [comment]
@ronykoz I like the idea. Let's also convert or create an example so we can verify that everything works end to end.

**Comment by ronykoz** [comment]
@glicht already tested it out, but sure will update some of our examples

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 3 above)
Comment (ronykoz) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:4140]: I think it's not necessary

File: Packs/ShiftManagement/ReleaseNotes/1_1_2.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+#### Scripts
+##### GetOnCallHoursPerUser
+- Updated the widget creation code 
+##### GetRolesPerShift
+- Updated the widget creation code 
+##### GetNumberOfUsersOnCall
+- Updated the widget creation code 
+##### GetShiftsPerUser
+- Updated the widget creation code 
+-->
```
Comment (yaakovi) [inline Packs/ShiftManagement/ReleaseNotes/1_1_2.md:11]: don't leave this empty, it looks bad in the UI.
`internal code improvements`

**Comment by kirbles19** [comment]
@ronykoz Done.

**Pull request merged**